chef/cookbooks/cinder/recipes/ceph.rb

```diff
@@ -17,36 +17,58 @@
 # Recipe:: ceph
 #
 
-glance_servers = search(:node, "roles:glance-server")
-if glance_servers.length > 0
-  glance_pool = glance_servers[0][:glance][:rbd][:store_pool]
-else   
-  glance_pool = nil
-end
+ceph_clients = {}
+ceph_keyrings = {}
 
 node[:cinder][:volumes].each_with_index do |volume, volid|
-  if volume['backend_driver'] == "rbd"
+  if volume[:backend_driver] == "rbd"
+
+    ceph_conf = volume[:rbd][:config_file]
+    admin_keyring = volume[:rbd][:admin_keyring]
 
-    check_ceph = Mixlib::ShellOut.new("ceph -k #{volume['rbd']['admin_keyring']} -c #{volume['rbd']['config_file']} -s | grep -q -e 'HEALTH_[OK|WARN]'")
+    check_ceph = Mixlib::ShellOut.new("ceph -k #{admin_keyring} -c #{ceph_conf} -s | grep -q -e 'HEALTH_[OK|WARN]'")
     check_ceph.run_command
 
     if check_ceph.exitstatus == 0
```

Comment:
Or maybe we should skip it only the first time? Once we have done the initial setup once, we could still put the backend in cinder.conf. Not totally sure?